### PR TITLE
LG-10766 - Authorization Page OIDC guide

### PR DESCRIPTION
**Pull request by nprimak**
https://cm-jira.usa.gov/browse/LG-10766

This is part of the dev doc redesign / rework with content updates and design updates broken down by page. This is for the Authorization page which has a few unique components as is a bit more complex design wise than the other pages.

When all page reworks are completed the navigation will be updated so that users can view the new pages. For now you must append /oidc/authorization to visit the page from home in Federalist preview. 

Biggest aspect of the rework is introduction of code snippets which scroll with the page and stay in the sidebar on the right hand side.

Figma can be seen here, please go to OIDC Guide section and Authorization page 
https://www.figma.com/file/3KSAr0qfDv8BXFIHyFEudZ/Login.gov-OIDC%2C-SAML%2C-and-UX-guides-(LG-10262%2C-LG-10514)?type=design&node-id=104-4001&mode=design

Note that the "return to top" button has been moved to the left so that it doesn't overlap with the gray code snippet sidebar and code snippets are hidden in mobile view.

### CODE REVIEW

File: assets/js/toggle_view.js
Comment (aduth) [inline assets/js/toggle_view.js:1]: I think it'd be a good idea to try to incorporate ESLint and Prettier into this project, using our [published ESLint configuration](https://www.npmjs.com/package/@18f/eslint-plugin-identity). We have this in `identity-site`, for example ([source](https://github.com/18F/identity-site/blob/d4e237a09aabcb23aacdc472c0a80f1df3053efe/.circleci/config.yml#L119-L127)).

A few issues I'm thinking that would help fix up here:

- [Naming convention](https://github.com/18F/identity-idp/blob/main/docs/frontend.md#naming-conventions) (camelCase)
- Whitespace

File: assets/js/toggle_view.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+// Elements for Listeners
+jwt_nav = document.getElementById("jwt-nav")
+pkce_nav = document.getElementById("pkce-nav")
+oidc_auth_success_button = document.getElementById("oidc_auth_success_button")
+oidc_auth_error_button = document.getElementById("oidc_auth_error_button")
+
+// Selectors
+oidc_auth_success_button.selector = "oidc_auth"
+oidc_auth_error_button.selector = "oidc_auth"
```
Comment (aduth) [inline assets/js/toggle_view.js:11]: If these are custom, I'd suggest using `dataset` (`data-` attributes) instead. These would be flagged by TypeScript if/when we integrate that into the project.

```suggestion
// Selectors
oidc_auth_success_button.dataset.selector = "oidc_auth"
oidc_auth_error_button.dataset.selector = "oidc_auth"
```

Better yet, maybe that can be added in the markup directly as `data-` attributes?

File: _pages/oidc/authorization.md [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+    <div class="grid-col-7">
+        The Authentication Context Class Reference requests can be used to specify the type of service level<sup id="fnref:1" role="doc-noteref"><a href="#fn:1" class="footnote" rel="footnote">1</a></sup> or the AAL (Authentication Assurance Level) for the user. These and the <code class="language-plaintext highlighter-rouge">scope</code> determine which <a class="usa-link" href="/attributes/">user attributes</a> will be available in the <a class="usa-link" href="oidc/#user-info-response">user info response</a>.
+      <p>
+        Multiple values can be joined with a space (before being URI-escaped in the final URL)
+      </p>
+    </div>
+  </div>
+  <div class="grid-row dev-doc-row">
+      <div class="usa-accordion">
+        {% include accordion.html content=type_of_service title="Type of Service Level" id="service_level" %}
```
[the remark this answers is not in the document]
Comment (nprimak) [reply _pages/oidc/authorization.md]: yes! I changed it so there is an` accordion_id` which is for toggling the accordion and then` id=""` in the include statement there corresponds to the `id` which you can use as a link

File: _pages/oidc/authorization.md [hunk 3]
```diff
@@ -0,0 +1,322 @@
+---
+title: OpenID Connect
+lead: >
+  [OpenID Connect](http://openid.net) is a simple identity layer built on top of the OAuth 2.0 protocol. Login.gov supports [version 1.0](http://openid.net/specs/openid-connect-core-1_0.html) of the specification and conforms to the [iGov Profile](https://openid.net/wg/igov).
```
Comment (karenstrother) [inline _pages/oidc/authorization.md:4]: Should there be external link notations for the links in the header?

**Comment by karenstrother** [comment]
From a content perspective this looks great! One comment about the external link markers.

File: assets/js/toggle_view.js
Comment (nprimak) [reply assets/js/toggle_view.js:1]: Created ticket and added to current sprint here: https://cm-jira.usa.gov/browse/LG-11480

**Comment by anniehirshman-gsa** [comment]
Adding a few very minor design/spacing suggestions (1/4): 

The "Success" and "Error" tabs have a small 6x right padding, which can be removed to match the left padding.

Screen Shot 2023-11-01 at 10 45 21 AM

**Comment by anniehirshman-gsa** [comment]
(2/4) Ideally it would be great to remove some spacing between the "Request parameters" heading and the tabs below it, so that the h3 is more closely associated with the content below it than the content above it.

Screen Shot 2023-11-01 at 10 48 11 AM

**Comment by anniehirshman-gsa** [comment]
(3/4) Could we add more space (2.5 rem) above the Next link, so that it stands out a little more as a separate navigational element (vs. within the last section of content)?

Screen Shot 2023-11-01 at 10 49 14 AM

**Comment by anniehirshman-gsa** [comment]
(4/4) The `acr_values` table has different spacing than the other table rows - ideally 2 rem at the top and bottom of this table element would be great.

Screen Shot 2023-11-01 at 10 51 29 AM

Thank you!!

**Review by Jeremy1026 — APPROVED** [review]

**Pull request merged**